File: python/agentic_commerce_protocol/models/allowance.py
# this file is @generated

from .allowance_reason import AllowanceReason
from .common import BaseModel


class Allowance(BaseModel):
    reason: AllowanceReason
    max_amount: int
    currency: str
    checkout_session_id: str
    merchant_id: str
    expires_at: str

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/Allowance.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Allowance(
    val reason: AllowanceReason,
    @SerialName("max_amount") val maxAmount: Long,
    val currency: String,
    @SerialName("checkout_session_id") val checkoutSessionId: String,
    @SerialName("merchant_id") val merchantId: String,
    @SerialName("expires_at") val expiresAt: String,
)
